examples/flax/image-captioning/run_image_captioning_flax.py

```diff
@@ -502,7 +504,8 @@ def main():
     # Get the column names for input/target.
     dataset_columns = image_captioning_name_mapping.get(data_args.dataset_name, None)
     if data_args.image_column is None:
-        assert dataset_columns is not None
+        if dataset_columns is None:
+            raise ValueError("Columns of dataset should not be None")
```

Comment:
`datasets_columns` is None if `data_args.dataset_name` is not in `image_captioning_name_mapping`:

https://github.com/huggingface/transformers/blob/f3a1efd1cf2e8b2aadec4a41d8bc8ef470de0275/examples/flax/image-captioning/run_image_captioning_flax.py#LL503

We should try and make the description for this ValueError reveal this problem! Could we state that `data_args.dataset_name` is not in `image_captioning_name_mapping` in the error message?

```suggestion
            raise ValueError(f"`--dataset_name` {data_args.dataset_name} not found in dataset '{data_args.dataset_name}'. Make sure to set `--dataset_name` to the correct dataset name, one of {', '.join(image_captioning_name_mapping.keys())}.")
```

(might need to run "make style" after incorporating these changes!)